### PR TITLE
Generate combined markdown docs file per build

### PR DESCRIPTION
**Pull request by ellismg**
Our current story for generating docs suitable for ingestion in
docs.microsoft.com is "generate a single markdown file with all the
commands and use anchors to cross reference". Since that's not
something the Cobra markdown formatters support out of the box, PC has
manually been merging these files together and doing some small
tweaks.

This PR attempts to automate that logic.

My first attempt at these changes had us to continue to use the Cobra
logic directly (by importing the package it was in) and then try to
clean up small things with some post processing. However, once I
realized that we would need to re-order the default output of the list
commands from SEE ALSO I decided to just take the existing code and
bring it in here and make small tweaks.

At some point, we will move to a different strategy where we have
multiple files and TOC that stitches them together, but until that
point this frees up a bunch of PC's time.

### CODE REVIEW

**Comment by ellismg** [comment]
@puicchan I applied the feedback from the content writer.  You can see the new generated documentation here:  https://gist.github.com/ellismg/d14ad642969588e39ffac8e9c868d780 

There are a few changes we couldn't adopt.

- When listing output formats, we use "json" instead of "JSON" as the  content writer suggested. This matches what is expected on the command line.

- When showing help for a command option with a default value, we print `(default: <value>)` at the end of the help text for that option. This behavior comes from the library we use for our CLI and is not easy to change.

There are a handful of remaining diffs, most of which are due to the same content being repeated twice and sourced from the same location (e.g. `provision` and `infra create` share the help text) and the content writer putting slightly different changes in each place. This was also common for global options. I've applied some level of judgement here, but in cases where there was a majority of support for one form, I adopted that.

There were some edits to the generated markdown that would be hard to bring back into the code (especially around URL formatting, where the markdown reads very poorly from a CLI). In these cases I did my best to make it look nice on the command line first and then to have a reasonable rendering in the generated markdown second.

You can do the comparison yourself, but I think this is probably the best we can do given the level of investment we're willing to put in here right now.

Let me know if you have any blocking concerns - otherwise I am going to get this in so we star the ball rolling on having automation here.

**Comment by puicchan** [comment]
Thanks! @ellismg . Can you update "## Azure Developer CLI"  to "## azd"?So the link for "Back to top" work? We actually ignored the content editor's suggestion and used azd in the end. And Savannah wants to have azd used more in content to boost SEO.

File: cli/azd/cmd/up.go [hunk 1]
```diff
@@ -36,9 +36,8 @@ This command executes the following in one step:
 	$ azd provision
 	$ azd deploy
 
-When no template is supplied, you can optionally select an azd template for cloning. Otherwise, "azd up" initializes the current directory so that your project is compatible with azd.`,
+When no template is supplied, you can optionally select an Azure Developer CLI template for cloning. Otherwise, running azd up initializes the current directory so that your project is compatible with Azure Developer CLI.`,
```
Comment (vhvb1989) [inline cli/azd/cmd/up.go:39]: use single quote for `azd up`

**Review by vhvb1989 — APPROVED** [review]
minor comments

File: cli/azd/cmd/init.go [hunk 2]
```diff
@@ -33,14 +33,13 @@ func initCmd(rootOptions *commands.GlobalCommandOptions) *cobra.Command {
 		},
 		rootOptions,
 		"init",
-		"Initialize a new application",
-		`Initialize a new application
+		"Initialize a new application.",
+		`Initialize a new application.
 
-When no template is supplied, you can optionally select an azd template for cloning. Otherwise, "azd init" initializes the current directory and creates resources so that your project is compatible with azd.
+When no template is supplied, you can optionally select an Azure Developer CLI template for cloning. Otherwise, `+"`"+`azd init`+"`"+`initializes the current directory and creates resources so that your project is compatible with Azure Developer CLI.
```
[the remark this answers is not in the document]
Comment (ellismg) [reply cli/azd/cmd/init.go:39]: I'll add `withBackticks()`, and then use it.

File: cli/azd/cmd/up.go (hunk 1 above)
Comment (ellismg) [reply cli/azd/cmd/up.go]: Good catch, thanks!

**Pull request merged**